compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -459,8 +459,11 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
 
                 for i in 0..dest_len {
                     let place = self.mplace_index(&dest, i)?;
-                    let value =
-                        if i == index { *elem } else { self.mplace_index(&input, i)?.into() };
+                    let value = if i == index {
+                        elem.clone()
```

Comment:
it's not going to matter much in the grand scheme of things by saving one (1) clone per call to this intrinsic, but this one should be easy to remove by calling `copy_op` in the branches ^^

Reply:
Yeah I considered this and decided, for the reasons you mentioned, that reducing code duplication is more important.